src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+import os
+import pathlib
+import tempfile
+import typing
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: typing.Dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.patches: typing.List[unittest.mock.patch] = []
+        for target, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
+            patch = unittest.mock.patch(target, mocked_path)
+            self.patches.append(patch)
+            patch.start()
+
+        motd_src = pathlib.Path(
+            f"{self.chroot}/etc/insights-client/insights-client.motd"
+        )
+        with motd_src.open("w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    @property
+    def chroot(self) -> str:
+        return self._dir.name
+
+    def __del__(self):
+        for patch in self.patches:
+            patch.stop()
+        self._dir.cleanup()
+
+
+@pytest.fixture(scope="function")
+def mock_fs():
+    fs = MockFS()
+    yield fs
+    del fs
+
+
+def test_present(mock_fs):
+    assert not os.path.exists(insights_client.MOTD_FILE)
+
+    # The file gets created/symlinked when .registered & .unregistered do not exist
+    insights_client.update_motd_message()
+    assert os.path.exists(insights_client.MOTD_FILE)
+    assert os.path.samefile(insights_client.MOTD_FILE, insights_client.MOTD_SRC)
+
+
+def test_absent_on_dot_registered(mock_fs):
+    # The MOTD file exists by default
+    insights_client.update_motd_message()
+    assert os.path.exists(insights_client.MOTD_FILE)
+
+    # It gets removed when .registered exists
+    with open(f"{mock_fs.chroot}/etc/insights-client/.registered", "w") as f:
+        f.write("")
+    insights_client.update_motd_message()
+    assert not os.path.exists(insights_client.MOTD_FILE)
+
+    # It stays absent when run multiple times
+    insights_client.update_motd_message()
+    assert not os.path.exists(insights_client.MOTD_FILE)
```

Comment:
This is a separate test case, verifying that the MOTD file stays absent if already absent and the `.registered` file exists.

```suggestion


def test_still_absent_on_dot_registered(mock_fs):
    with open(f"{mock_fs.chroot}/etc/insights-client/.registered", "w") as f:
        f.write("")
    insights_client.update_motd_message()
    assert not os.path.exists(insights_client.MOTD_FILE)
```

If the test names were action based, it could be called something like “test_not_created_on_registered”.

Reply:
I wouldn't say it's a separate case. I've updated the separation of blocks to make it more clear.